util/codegen/codegen.go

```diff
@@ -85,28 +85,35 @@ func NewImportTracker(thisPkg *types.Package) *ImportTracker {
 	}
 }
 
+type namePkgPath struct {
+	name    string // optional import name
+	pkgPath string
+}
+
 // ImportTracker provides a mechanism to track and build import paths.
 type ImportTracker struct {
 	thisPkg  *types.Package
-	packages map[string]bool
+	packages map[namePkgPath]bool
 }
 
-func (it *ImportTracker) Import(pkg string) {
-	if pkg != "" && !it.packages[pkg] {
-		mak.Set(&it.packages, pkg, true)
+// Import imports pkgPath under an optional import name.
```

Comment:
I found this a little confusing to read at the call site, when the target name is blank. What do you think about using a separate method:

```go
func (it *ImportTracker) ImportWithAlias(alias, pkgPath string) { …
```
perhaps?

Reply:
I think it's okay as is. If we add `ImportWithAlias`, it also calls into question whether to also add `HasWithAlias`, but then the semantics of `Has` becomes ambiguous whether you're simply checking whether the pkgPath has been imported at all, or imported without being renamed.

This package isn't widely used and if it becomes a sharp edge, we can refactor the callsites in the future.